testblas/level1double.go

```diff
@@ -1092,6 +1092,9 @@ func DdotTest(t *testing.T, d Ddotter) {
 	if n := ddot(4, x[:4], 1, x[1:], 1); n != 4 {
 		t.Errorf("ddot: mismatch Unaligned: expected %v, found %v", 4, n)
 	}
+	if n := ddot(2, x[:4], 2, x[1:], 2); n != 2 {
```

Comment:
Could you add an odd-stride test as well? That was how I discovered it in Dgemm. Thanks!